LDAR_Sim/src/virtual_world/nonfugitive_emissions.py

```diff
@@ -0,0 +1,163 @@
+from typing import Any
+from math import ceil
+from typing_extensions import override
+from virtual_world.emissions import Emission
+from file_processing.output_processing.output_utils import EmisInfo
+
+
+EXPIRE = "expired"
+
+
+class NonRepairableEmission(Emission):
+    def __init__(
+        self,
+        emission_n: int,
+        rate: float,
+        start_date: date,
+        simulation_sd: date,
+        repairable: bool,
+        tech_spat_cov_probs: dict[str, float],
+        duration: int,
+    ):
+        super().__init__(
+            emission_n,
+            rate,
+            start_date,
+            simulation_sd,
+            repairable,
+            tech_spat_cov_probs,
+        )
+        self._tagged: bool = False
+        self._days_since_tagged: int = 0
+        self._tagged_by_company: str = None
+        self._tagged_by_crew: str = None
+        self._tagging_rep_delay: int = 0
+        self._expiry_date: date = None
+        self._duration: int = duration
+        days_active_b4_sim: int = (simulation_sd - start_date).days
+        self._days_active_b4_sim = days_active_b4_sim if days_active_b4_sim > 0 else 0
+
+    def __reduce__(self):
+        return self._reconstruct_nonfugitive_emission, (self.__dict__,)
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+
+    @classmethod
+    def _reconstruct_nonfugitive_emission(cls, state):
+        instance = cls.__new__(cls)
+        instance.__setstate__(state)
+        return instance
+
+    def get_init_detect_company(self):
+        return self._init_detect_by
+
+    def tagged_today(self) -> bool:
+        return (
+            self._tagged and (self._days_since_tagged == 0)
+        ) and self._tagged_by_company != EXPIRE
+
+    def get_daily_emissions(self) -> float:
+        return self._rate * 86.4
+
+    def expire(self, emis_info: EmisInfo):
+        """
+        Checks the days since start against the duration
+        Ends the emission if possible
+        """
+        self._tagged = True
+        self._tagged_by_company = EXPIRE
+        self._status = EXPIRE
+        emis_info.emis_expired += 1
+        self._repair_date = self._start_date + timedelta(days=(self._active_days))
+
+    # TODO potentially move this into company later
+    def estimate_start_date(self, cur_date: date, t_since_ldar: int) -> None:
+        """Estimates the start date and days activate of the fugitive emission based on
+        the time since the last LDAR-SIm and the current date (discovery date)
+
+        Args:
+            cur_date (date): The current date in the simulation
+            t_since_ldar (int): THe time in days since the site at which the emissions
+            was discovered last received LDAR
+        """
+        half_duration: int = ceil((t_since_ldar / 2))
+        self._estimated_days_active = half_duration
+        self._estimated_date_began = cur_date - timedelta(days=half_duration)
+
+    def tag_leak(
```

Comment:
tag_leak tag_leak should not need to be implemented for non-repairable emissions